src/test/regress/expected/subquery_prepared_statements.out

```diff
@@ -0,0 +1,287 @@
+-- ===================================================================
+-- test recursive planning functionality on prepared statements
+-- ===================================================================
+CREATE SCHEMA subquery_prepared_statements;
+SET search_path TO subquery_prepared_statements, public;
+CREATE TYPE xy AS (x int, y int);
+SELECT run_command_on_workers('CREATE SCHEMA subquery_prepared_statements');
+       run_command_on_workers        
+-------------------------------------
+ (localhost,57637,t,"CREATE SCHEMA")
+ (localhost,57638,t,"CREATE SCHEMA")
+(2 rows)
+
+SELECT run_command_on_workers('CREATE TYPE subquery_prepared_statements.xy AS (x int, y int)');
+      run_command_on_workers       
+-----------------------------------
+ (localhost,57637,t,"CREATE TYPE")
+ (localhost,57638,t,"CREATE TYPE")
+(2 rows)
+
+SET client_min_messages TO DEBUG1;
+PREPARE subquery_prepare_without_param AS 
+SELECT
+   DISTINCT values_of_subquery
+FROM
+    (SELECT 
+      DISTINCT (users_table.user_id, events_table.event_type)::xy  as values_of_subquery
+     FROM 
+      users_table, events_table 
+     WHERE 
+      users_table.user_id = events_table.user_id AND 
+     event_type IN (1,2,3,4)
+     ORDER BY 1 DESC LIMIT 5
+     ) as foo
+    ORDER BY 1 DESC;
+PREPARE subquery_prepare_param_on_partkey(int) AS 
+SELECT
+   DISTINCT values_of_subquery
+FROM
+    (SELECT 
+      DISTINCT (users_table.user_id, events_table.event_type)::xy  as values_of_subquery
+     FROM 
+      users_table, events_table 
+     WHERE 
+      users_table.user_id = events_table.user_id AND
+      (users_table.user_id = $1 OR users_table.user_id = 2) AND
+     event_type IN (1,2,3,4)
+     ORDER BY 1 DESC LIMIT 5
+     ) as foo
+    ORDER BY 1 DESC;
+PREPARE subquery_prepare_param_non_partkey(int) AS 
+SELECT
+   DISTINCT values_of_subquery
+FROM
+    (SELECT 
+      DISTINCT (users_table.user_id, events_table.event_type)::xy  as values_of_subquery
+     FROM 
+      users_table, events_table 
+     WHERE 
+      users_table.user_id = events_table.user_id AND 
+     event_type = $1
+     ORDER BY 1 DESC LIMIT 5
+     ) as foo
+    ORDER BY 1 DESC;
+-- execute each test with 6 times
+EXECUTE subquery_prepare_without_param;
+DEBUG:  push down of limit count: 5
+DEBUG:  generating subplan 1_1 for subquery SELECT DISTINCT ROW(users_table.user_id, events_table.event_type)::subquery_prepared_statements.xy AS values_of_subquery FROM public.users_table, public.events_table WHERE ((users_table.user_id = events_table.user_id) AND (events_table.event_type = ANY (ARRAY[1, 2, 3, 4]))) ORDER BY ROW(users_table.user_id, events_table.event_type)::subquery_prepared_statements.xy DESC LIMIT 5
+ values_of_subquery 
+--------------------
+ (6,4)
+ (6,3)
+ (6,2)
+ (6,1)
+ (5,4)
+(5 rows)
+
+EXECUTE subquery_prepare_without_param;
+DEBUG:  push down of limit count: 5
```

Comment:
It is confusing to me that we're planning this again. @marco any insight on what's going on?

Reply:
Could be that the prepared statement cache is getting invalidated.

(I wonder whether DROP SCHEMA aggressively invalidates everything)